analyzers/tests/SonarAnalyzer.UnitTest/AnalysisContext/SonarCompilationStartAnalysisContextTest.Register.cs

```diff
@@ -0,0 +1,64 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2022 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Linq.Expressions;
+using Moq;
+using SonarAnalyzer.AnalysisContext;
+
+namespace SonarAnalyzer.UnitTest.AnalysisContext
+{
+    public partial class SonarAnalysisContextTest
+    {
+        [TestMethod]
+        public void SonarCompilationStartAnalysisContext_RegisterCompilationEndAction() =>
+            TestStartContextRegistration<SonarCompilationReportingContext>(
+                registrationSetup: x => x.RegisterCompilationEndAction(It.IsAny<Action<CompilationAnalysisContext>>()),
+                registration: (context, action) => context.RegisterCompilationEndAction(action));
+
+        [TestMethod]
+        public void SonarCompilationStartAnalysisContext_RegisterSemanticModel()
+        {
+            TestStartContextRegistration<SonarSematicModelReportingContext>(
+                registrationSetup: x => x.RegisterSemanticModelAction(It.IsAny<Action<SemanticModelAnalysisContext>>()),
+                registration: (context, action) => context.RegisterSemanticModelAction(action));
+        }
+
+        [TestMethod]
+        public void SonarCompilationStartAnalysisContext_RegisterSymbolAction() =>
+            TestStartContextRegistration<SonarSymbolReportingContext>(
+                registrationSetup: x => x.RegisterSymbolAction(It.IsAny<Action<SymbolAnalysisContext>>(), It.IsAny<ImmutableArray<SymbolKind>>()),
+                registration: (context, action) => context.RegisterSymbolAction(action));
```

Comment:
I could not test the `x.RegisterNodeAction` because it behaves quite differently from the others by creating its own context and ignoring the one that is setup by Moq.